relay-metrics/src/aggregation.rs

```diff
@@ -786,6 +805,30 @@ impl BucketKey {
         std::hash::Hash::hash(self, &mut hasher);
         hasher.finalize() as i64
     }
+
+    /// Estimates the number of bytes needed to encode the bucket key.
+    /// Note that this does not necessarily match the exact memory footprint of the key,
+    /// because datastructures might have a memory overhead.
+    fn cost(&self) -> usize {
+        // We use private constants here to make sure that a change in the implementation
+        // does not change the cost model without being noticed.
+        // Note that when the actual data type of any of the subkeys changes, we have to update
+        // the cost model and potentially increase the limits that are applied to it.
+        const PROJECT_KEY_SIZE: usize = 32;
+        const TIMESTAMP_SIZE: usize = mem::size_of::<u64>();
+        const METRIC_TYPE_SIZE: usize = mem::size_of::<u8>(); // small enum
+        const METRIC_UNIT_SIZE: usize = 16; // 1 (discriminator) + 15 (size of CustomUnit, the largest value)
```

Comment:
Again, here please use `size_of::<MetricUnit>()`. There's even [a test](https://github.com/getsentry/relay/blob/bff235eecd595c9daad1c8d8e0dca14fd51c3aa4/relay-metrics/src/protocol.rs#L520) asserting this, but no need to introduce more magic numbers.

Same for the others. Please always use `size_of()`. In fact, you could then get rid of the constants (because it's a const fn anyway), and add them directly.